Develop

Tempestmon opened the pull request.

Ramblball committed:
fix user table name

Ramblball committed:
add .gitignore

Ramblball committed:
update eventor_schema

Ramblball committed:
update User structure

Ramblball committed:
update Event structure

Ramblball committed:
update Category structure

change category from class to enum

Ramblball committed:
Merge branch 'feature_database_structure' into develop

Ramblball committed:
update project sources

Ramblball committed:
update interaction with user

Ramblball committed:
update interaction with event

Ramblball committed:
Merge branch 'feature_DAO_Services' into develop

Tempestmon committed:
make setPassword public

Tempestmon committed:
simple view

Tempestmon committed:
simple controller

Tempestmon committed:
Merge branch 'feature_View' into develop

Tempestmon committed:
Merge branch 'feature_Controller' into develop

Ramblball committed:
update resouces

Ramblball committed:
comment password confirming

Ramblball committed:
update id generation and comment password confirming

Tempestmon committed:
update view and control

Ramblball committed:
fix warnings

update Controller modifier and add escape condition for a while(true)

Ramblball committed:
fix checkPassword method

now use java.Util.Arrays.equals to compare hashes

Ramblball committed:
add password confirming

Ramblball committed:
remove reduced code

Ramblball committed:
allocate a method for opening sessions

Ramblball committed:
add javaDoc to services

Ramblball committed:
allocate string literals to one class

Ramblball committed:
replace all JUnit dependencies with aggregator

Ramblball committed:
Merge branch 'fix' into develop

Ramblball committed:
update README.md and pom.xml

Ramblball committed:
fix README.md

Tempestmon committed:
create constant keywords

Tempestmon committed:
replace concatenation by StringBuilder

Tempestmon committed:
split Controller

Tempestmon committed:
add JavaDoc to Controller

Tempestmon committed:
Merge branch 'fix' into develop

Tempestmon committed:
Add view error processing of arguments + formatting

Ramblball committed:
fix sequence exception

Ramblball committed:
update add error handling

Ramblball committed:
update validation and error handling

Ramblball committed:
Merge branch 'fix_validation_and_error_handling' into develop

Ramblball committed:
minor fixes

Tempestmon committed:
make echo bot

Tempestmon committed:
make longpoll bot

Tempestmon committed:
Merge branch 'telegram_bot' into develop

Tempestmon committed:
create user controller tests

Tempestmon committed:
create event controller tests

Tempestmon committed:
Merge branch 'controller_test' into develop

Tempestmon committed:
create chat tests, make execute public for tests, formatting

Tempestmon committed:
Merge branch 'view_test' into develop

Ramblball committed:
update project configuration

Ramblball committed:
add query class for search with criteria

Ramblball committed:
update entities

Ramblball committed:
update DAO and service for search

Ramblball committed:
Controller refactor

Ramblball committed:
Merge branch 'large_update_sry' into develop

Tempestmon committed:
create buttons for chat

Tempestmon committed:
Merge branch 'telegram' into develop

Tempestmon committed:
Merge remote-tracking branch 'origin/develop' into develop

Ramblball committed:
update event entity

Ramblball committed:
update DAO

allocate interaction with event to EventDAO

Ramblball committed:
update services

allocate event interaction, add crud operations and add exception handling

Ramblball committed:
update Controller

allocate current user info to abstract class, add user interaction with event and add date support

Ramblball committed:
Complete query

update tests and add search by date support

Ramblball committed:
minor changes

Ramblball committed:
Merge branch 'interaction_and_minor_updates' into develop

Tempestmon committed:
attach new functions to buttons + formatting

Tempestmon committed:
update help

Ramblball committed:
add tests for EventServices

Ramblball committed:
add tests for controllers

Ramblball committed:
Merge branch 'add_tests' into develop

Tempestmon committed:
hide token

Tempestmon committed:
generate new token

Tempestmon committed:
generate new token

eez-eh commented on README.md:
```diff
@@ -1 +1,40 @@
-Eventor
+# Eventor
+
```
:+1:

eez-eh commented on src/main/java/view/Chat.java:
```diff
@@ -0,0 +1,153 @@
+    private final EventController eventController = new EventController();
+    private final static String fewArgumentsOutput = "Too few arguments. Try to type \"help\"";
+
+    public String execute(String message) {
```
Метод перенасыщен функционалом.

Добавить отдельную сущность "Команда" с методом execute() и наследоваться от неё классами различных команд.

В этом же методе работать с абстракциями, принимать на вход объект "Сообщение" с полями userId, text и другими.

Помимо этого, добавить в этот класс сущность TelegramBot, у которой мы и будем дёргать метод для получения ответа на какую-либо команду (принимать на вход будем объект нового класса "Сообщение")

Название этого класса скорректировать, к примеру, TelegramProvider.

eez-eh commented on src/main/java/view/Chat.java:
```diff
@@ -0,0 +1,153 @@
+        List<InlineKeyboardButton> fifthRow = new ArrayList<>();
+        List<InlineKeyboardButton> sixthRow = new ArrayList<>();
+
+
```
Убрать пустую строку.

eez-eh commented on src/main/java/view/Chat.java:
```diff
@@ -0,0 +1,153 @@
+
+    @Override
+    public String getBotUsername() {
+        return "eventor_oop_bot";
```
Вынести в константное поле класса

eez-eh commented on src/main/java/view/Chat.java:
```diff
@@ -0,0 +1,153 @@
+    }
+
+    @Override
+    public String getBotToken() {
```
:+1:

eez-eh commented on src/main/java/view/Console.java:
```diff
@@ -1,4 +1,65 @@
+
+    public void execute() {
+        Scanner scanner = new Scanner(System.in);
+        while (true) {
```
Замечания аналогичны реализации класса Chat

eez-eh commented on src/main/java/controller/Controller.java:
```diff
@@ -0,0 +1,16 @@
+
+import database.model.User;
+
+public abstract class Controller {
```
Тут и далее JavaDoc на все классы и интерфейсы.
JavaDoc-и просьба формулировать на русском.

eez-eh commented on src/main/java/controller/Controller.java:
```diff
@@ -0,0 +1,16 @@
+
+    /**
+     * Check that user is not authorized
+     *
```
Тут и далее убрать подобные пустые строки

eez-eh commented on src/main/java/controller/EventController.java:
```diff
@@ -0,0 +1,185 @@
+    private StringBuilder stringBuilder = new StringBuilder();
+
+    /**
+     * @return manual
```
Описать возвращаемый параметр

eez-eh commented on src/main/java/controller/EventController.java:
```diff
@@ -0,0 +1,185 @@
+import java.util.List;
+import java.util.StringJoiner;
+
+public class EventController extends Controller {
```
ditto

eez-eh commented on src/main/java/controller/EventController.java:
```diff
@@ -0,0 +1,185 @@
+    /**
+     * Validate events params before save
+     *
+     * @param name        Name of event
```
ditto

eez-eh commented on src/main/java/controller/EventController.java:
```diff
@@ -0,0 +1,185 @@
+    /**
+     * Create new event
+     *
+     * @param name        of new event
```
`        of new event` -> `название события`

eez-eh commented on src/main/java/view/Chat.java:
```diff
@@ -0,0 +1,153 @@
+        }
+    }
+
+    private InlineKeyboardMarkup setKeyboard() {
```
JavaDoc на все private методы классов и public методы интерфейсов

eez-eh commented on src/main/java/view/Buttons.java:
```diff
@@ -0,0 +1,19 @@
+import org.telegram.telegrambots.meta.api.objects.replykeyboard.buttons.InlineKeyboardButton;
+
+public final class Buttons {
+    public final static InlineKeyboardButton createUser = new InlineKeyboardButton().setText("Create user").setSwitchInlineQueryCurrentChat("create user ");
```
Однострочный комментарий для каждой константы

eez-eh commented on src/main/java/database/utils/QueryLiterals.java:
```diff
@@ -0,0 +1,22 @@
+package database.utils;
+
+public class QueryLiterals {
```
Однострочный комментарий для каждой константы

eez-eh commented on src/main/java/database/type/EnumTypePostgreSQL.java:
```diff
@@ -0,0 +1,26 @@
+            st.setObject( index, value.toString(), Types.OTHER );
+        }
+    }
+
```
Удалить лишние строки ниже

eez-eh commented on src/main/java/controller/Keywords.java:
```diff
@@ -0,0 +1,35 @@
+
+    public final static String dateTimeFormat = "yyyy-MM-dd HH:mm";
+
+    public final static String help = "This is a bot for creating events.\nTo create a user, type: \"create user *name* *password*\"\nTo log in, type: \"login *name* *password*\"\nTo create an event, type:\n \"create event *name* *time:(yyyy-MM-dd HH:mm)* *place* *description*\"\nTo find an event by name, type: \"find *eventName\"\nTo subscribe for an event, type: \"signup *event id*\"\nTo unsubs ...
```
Предлагаю не выносить столь длинную строку в константы
Лучше заменить на несколько print-ов в том месте, где константа используется

eez-eh commented on src/test/java/dbTests/ModelTests.java:
```diff
@@ -0,0 +1,32 @@
+
+    @Test
+    public void test() {
+//        Event event = new Event("");
```
Убрать закомментированный код

eez-eh replied:
Всё ещё актуально

eez-eh reviewed (commented):
toFix

Tempestmon committed:
create commands, telegramProvider, telegramChat, operations, message

Tempestmon committed:
refactor console

Tempestmon committed:
Java-docs + commentaries

Tempestmon committed:
Merge branch 'view_refactor' into develop

Ramblball committed:
update methods

Tempestmon committed:
refactor(view): использование имени пользователя из tg

Ramblball committed:
refactor(controller, model): удаление функционала авторизации

Ramblball committed:
refactor(controller, model):  Перенесение функционала подписки/отписки

Tempestmon committed:
refactor(view): Удаление функционала авторизации

Ramblball committed:
docs(all): документирование классов приложения

Ramblball committed:
feature(all): добавление получения мероприятий

Ramblball committed:
chore(all): добавление логера в pom.xml

Tempestmon committed:
refactor(view): декомпозирование parseToMessage в Provider

Ramblball committed:
feature,refactor(controller, model): добавление обработки ошибок

Tempestmon committed:
refactor(view): декомпозирование диалогов в DialogTransmitter

Tempestmon committed:
fix(view): одновременная работа с пользователями

Tempestmon committed:
refactor(view): добавление обработки неизвестной команды

Tempestmon committed:
docs(view): документирование TelegramBot и DialogTransmitter

Ramblball committed:
feature, refactor, doc(controller): логирование ошибок

eez-eh commented on src/main/resources/Hibernate-2020_11_26_07_44_50-dump.sql:
```diff
@@ -0,0 +1,239 @@
+-- Dumped from database version 12.5 (Ubuntu 12.5-0ubuntu0.20.04.1)
+-- Dumped by pg_dump version 12.5 (Ubuntu 12.5-0ubuntu0.20.04.1)
+
+SET statement_timeout = 0;
```
В рамках следующей задачи предлагаю добавить миграции при помощи flyway

eez-eh commented on src/test/java/controllerTests/EventControllerTests.java:
```diff
@@ -0,0 +1,205 @@
+import static org.junit.jupiter.api.Assertions.*;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class EventControllerTests {
```
Почему весь код закомментирован?
Нам нужен этот класс?

eez-eh commented on src/test/java/controllerTests/UserControllerTests.java:
```diff
@@ -0,0 +1,112 @@
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class UserControllerTests {
+//    UserController controller;
```
ditto

eez-eh commented on src/test/java/dbTests/EventQueryTest.java:
```diff
@@ -0,0 +1,85 @@
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class EventQueryTest {
+//
```
ditto

eez-eh commented on src/test/java/dbTests/services/EventServiceTest.java:
```diff
@@ -0,0 +1,190 @@
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class EventServiceTest {
+//    EventService service;
```
ditto

eez-eh commented on src/test/java/viewTests/ProviderTests.java:
```diff
@@ -0,0 +1,55 @@
+//import org.junit.jupiter.api.Test;
+//import view.TelegramProvider;
+//
+//import static org.junit.jupiter.api.Assertions.*;
```
ditto

eez-eh commented on token.txt:
```diff
@@ -0,0 +1 @@
+1409038970:AAHKwqp0dZcajwJOcYlH6j6irBcsQcXyoEM
```
Все секретные токены и пароли никогда не должны появляться в git-е.
Иначе спаммеры просканируют весь github, соберут токены и начнут слать вашим пользователям спам от имени вашего бота. Все секретные ключи и токены можно держать в переменных среды (Environment variables) или во внешних файлах, которые добавлены в .gitignore файл.

eez-eh commented on src/main/java/view/commands/Unsubscribing.java:
```diff
@@ -0,0 +1,16 @@
+/**
+ * Команда отписки от мероприятия
+ */
+public class Unsubscribing extends Command {
```
`Unsubscribing` -> `UnsubeCommand`

eez-eh commented on src/main/java/view/commands/Unknown.java:
```diff
@@ -0,0 +1,13 @@
+public class Unknown extends Command {
+    @Override
+    public String execute(Message message) {
+        return "Unknown command. Try to type \"help\"";
```
Тут лучше использовать форматированную строку и передавать её в метод format: String.format("Unknown command %s. Try to type \"help\"", передать команду, введённую юзером);

Тогда можно вынести и "Unknown command %s. Try to type \"help\"" в класс строковых сообщений

eez-eh commented on src/main/java/view/commands/UpdatingEvent.java:
```diff
@@ -0,0 +1,17 @@
+/**
+ * Команда обновления мероприятия
+ */
+public class UpdatingEvent extends Command {
```
UpdateEventCommand

eez-eh commented on src/main/java/view/commands/Subscribing.java:
```diff
@@ -0,0 +1,16 @@
+/**
+ * Команда подписки на мероприятие
+ */
+public class Subscribing extends Command {
```
SubscribeCommand

eez-eh commented on src/main/java/view/commands/RemovingEvent.java:
```diff
@@ -0,0 +1,16 @@
+/**
+ * Команда удаления мероприятия
+ */
+public class RemovingEvent extends Command {
```
RemoveEventCommand

eez-eh commented on src/main/java/view/commands/Help.java:
```diff
@@ -0,0 +1,17 @@
+/**
+ * Команда вызова помощи
+ */
+public class Help extends Command {
```
HelpCommand

eez-eh commented on src/main/java/view/commands/GetSubscribes.java:
```diff
@@ -0,0 +1,16 @@
+/**
+ * Команда для получения мероприятий, на которые пользователь подписан
+ */
+public class GetSubscribes extends Command{
```
SubscribesGetCommand

eez-eh commented on src/main/java/view/commands/GetOwnEvents.java:
```diff
@@ -0,0 +1,16 @@
+/**
+ * Команда для получения мероприятий созданных пользователем
+ */
+public class GetOwnEvents extends Command{
```
OwnEventsGetCommand

eez-eh commented on src/main/java/view/commands/FindingEventParameters.java:
```diff
@@ -0,0 +1,22 @@
+/**
+ * Команда создания мероприятия с критериями
+ */
+public class FindingEventParameters extends Command {
```
EventParametersFindCommand

eez-eh commented on src/main/java/view/commands/FewArguments.java:
```diff
@@ -0,0 +1,13 @@
+public class FewArguments extends Command {
+    @Override
+    public String execute(Message message) {
+        return "Too few arguments. Try to type \"help\"";
```
Тоже использовать форматированные строки
Пример описал ниже

eez-eh commented on src/main/java/view/commands/EventCreating.java:
```diff
@@ -0,0 +1,16 @@
+/**
+ * Команда создания мероприятия
+ */
+public class EventCreating extends Command {
```
ditto

eez-eh commented on src/main/java/view/commands/Command.java:
```diff
@@ -0,0 +1,10 @@
+/**
+ * Общий класс описывающий логику команд
+ */
+public abstract class Command {
```
Добавить интерфейс ICommand

eez-eh replied:
Описать в нём метод execute

eez-eh commented on src/main/java/view/TelegramBot.java:
```diff
@@ -0,0 +1,94 @@
+    @Override
+    public void onUpdateReceived(Update update) {
+        if (update.hasMessage() && update.getMessage().hasText()) {
+            String address = "@" +
```
Использовать String.format

eez-eh commented on src/main/java/view/TelegramBot.java:
```diff
@@ -0,0 +1,94 @@
+
+    /**
+     * Получает сообщение от пользователя, отдаёт на обработку Provider`у и отправляет ответ
+     *
```
Убрать пустую строку

eez-eh replied:
Выше везде тоже

eez-eh commented on src/main/java/Main.java:
```diff
@@ -1,10 +1,24 @@
+        } catch (TelegramApiException e) {
+            e.printStackTrace();
+        }
+//        var console = new Console();
```
Убрать закомментированный код

eez-eh commented on src/main/java/controller/Controller.java:
```diff
@@ -0,0 +1,29 @@
+    /**
+     * Возвращает информацию о пользователе, работающем с ботом
+     * @param id        Id пользователя
+     * @return            Объект текущего пользователя
```
Неровное форматирование относительно предыдущей строки

eez-eh commented on src/main/java/controller/Controller.java:
```diff
@@ -0,0 +1,29 @@
+
+    /**
+     * Возвращает информацию о пользователе, работающем с ботом
+     * @param id        Id пользователя
```
`Id пользователя` -> `уникальный идентификатор пользователя`

eez-eh commented on src/main/java/controller/Controller.java:
```diff
@@ -0,0 +1,29 @@
+     * @param id        Id пользователя
+     * @return            Объект текущего пользователя
+     */
+    public User getCurrent(Integer id) throws DBException, NotAuthorizedException {
```
:+1:

eez-eh commented on src/main/java/controller/Controller.java:
```diff
@@ -0,0 +1,29 @@
+    public User getCurrent(Integer id) throws DBException, NotAuthorizedException {
+        User user = userService.findById(id);
+        if (user == null) {
+            throw new NotAuthorizedException(Keywords.notAuth);
```
:+1:

eez-eh commented on src/main/java/controller/EventController.java:
```diff
@@ -0,0 +1,278 @@
+public class EventController extends Controller {
+
+    /**
+     * Возвращает подсказку по оформлению команд для пользователя
```
Метод для получения подсказки по ...

eez-eh commented on src/main/java/controller/EventController.java:
```diff
@@ -0,0 +1,278 @@
+
+    /**
+     * Возвращает подсказку по оформлению команд для пользователя
+     * @return              Помощь
```
справка по командам приложения

eez-eh commented on src/main/java/controller/EventController.java:
```diff
@@ -0,0 +1,278 @@
+     * @param id            Id пользователя
+     * @return              Список мероприятий
+     */
+    public String getSubs(Integer id) {
```
`getSubs` - давайте будем избегать сокращений слов (ухудшает читаемость)
Не понятно становится, что имелось в виду - getSubscribes или же getSubscribers или же getSubscriptions

eez-eh commented on src/main/java/controller/Keywords.java:
```diff
@@ -0,0 +1,56 @@
+    ----------------- Успешное выполнение --------------------------------
+     */
+    //Пользователь создан
+    public final static String userCreated = "Добро пожаловать, '%s'";
```
Для констант принята немного иная нотация в java: https://www.oracle.com/java/technologies/javase/codeconventions-namingconventions.html
`userCreated` -> `USER_CREATED`
Переименуйте пожалуйста тут и далее

eez-eh commented on src/main/java/controller/UserController.java:
```diff
@@ -0,0 +1,48 @@
+     * @param tgUser      Объект пользователя в телеграмме
+     * @return            Результат обновления
+     */
+    public String update(org.telegram.telegrambots.meta.api.objects.User tgUser) {
```
org.telegram.telegrambots.meta.api.objects.User вынести в импорты

eez-eh commented on src/main/java/controller/Validator.java:
```diff
@@ -0,0 +1,45 @@
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public class Validator {
```
JavaDoc

eez-eh commented on src/main/java/controller/Validator.java:
```diff
@@ -0,0 +1,45 @@
+     * @throws ParseException       Неверный формат даты
+     * @throws PastTimeException    Введенная дата прошла
+     */
+    public void checkEventsParams(String name, String time, String place, String description) throws ValidationException {
```
`checkEventsParams` -> `checkEventParams`

eez-eh commented on src/main/java/controller/exception/LongFieldException.java:
```diff
@@ -0,0 +1,11 @@
+package controller.exception;
+
+public class LongFieldException extends ValidationException {
```
JavaDoc

eez-eh commented on src/main/java/controller/exception/NotAuthorizedException.java:
```diff
@@ -0,0 +1,11 @@
+package controller.exception;
+
+public class NotAuthorizedException extends Exception {
```
JavaDoc

eez-eh commented on src/main/java/controller/exception/ParseException.java:
```diff
@@ -0,0 +1,11 @@
+package controller.exception;
+
+public class ParseException extends ValidationException {
```
JavaDoc

eez-eh commented on src/main/java/controller/exception/PastTimeException.java:
```diff
@@ -0,0 +1,11 @@
+package controller.exception;
+
+public class PastTimeException extends ValidationException {
```
JavaDoc

eez-eh commented on src/main/java/controller/exception/ValidationException.java:
```diff
@@ -0,0 +1,11 @@
+package controller.exception;
+
+public abstract class ValidationException extends Exception {
```
JavaDoc

eez-eh commented on src/main/java/database/DBException.java:
```diff
@@ -0,0 +1,11 @@
+package database;
+
+public class DBException extends Exception{
```
ditto

eez-eh replied:
Тут и далее вынести в пакет database.exception

eez-eh commented on src/main/java/view/Provider.java:
```diff
@@ -0,0 +1,30 @@
+    private static final HashMap<String, Command> hashMap = new HashMap<>();
+
+    Provider(){
+        hashMap.put("Создать", new EventCreating());
```
Отдельный Provider не нужен, сделать метод setCommands в классе, где это вызывается и инициализировать мапу в конструкторе

eez-eh commented on src/main/java/view/DialogTransmitter.java:
```diff
@@ -0,0 +1,217 @@
+        }
+        createKeyboard();
+        message.setUser(user);
+        switch (received) {
```
Тут тоже нужна аналогичная мапа private static final HashMap<String, Command> hashMap = new HashMap<>(); для команд

eez-eh reviewed (commented):
lgtm
В целом всё хорошо, до сдачи можете ничего не менять, pull request прошу пока не закрывать.
Закрыть и отправить изменения из develop в master только тогда, когда исправите остальные замечания.
FYI: ожидаю, что код готовы будут показать оба участника команды

Ramblball committed:
docs,style(controller,model): исправление обсуждений

Удаление неиспользуемого кода. Исправление javaDoc. Выравнивание кода. Исправление названий констант.

Ramblball committed:
fix, chore(app): обновление log4j

Tempestmon committed:
refactor(view): исправление имён команд

Tempestmon committed:
refactor(view): исправление имён команд

Tempestmon committed:
refactor(view): исправление имён команд
refactor(view): исправление имён команд

Tempestmon committed:
Merge remote-tracking branch 'origin/develop' into develop

Tempestmon committed:
refactor(view): добавление интерфейса команд

Tempestmon committed:
refactor(view): устранение провайдера

Tempestmon committed:
refactor(view): добавление отдельного класса для клавиатуры телеграмм бота TelegramKeyboard

eez-eh commented:
lgtm

eez-eh commented:
Единственное, несколько некритичных замечаний не были поправлены.
Прошу проверить и поправить в Task 3.
pull request можно закрыть.

Ramblball committed:
refactor, feature(view): добавление параллельной работы пользователей

Ramblball committed:
feature, refactor(app): вывод мероприятий за текущую неделю

Tempestmon committed:
feature(model): добавление перехода в диалог с организатором мероприятия

Tempestmon committed:
feature(view): добавление emoji, возможности управления с помощью emoji

Tempestmon committed:
feature(view): добавление функции сохранения отформатированного текста

Tempestmon committed:
fix(view): устранение форматирования времени

Tempestmon committed:
refactor(view): декомпозиция веток диалога из DialogTransmitter с помощью Answer

Tempestmon committed:
fix(view): возможность сохранять введённые данные пользователя

Tempestmon committed:
refactor, docs(view): документирование Answers, CommandMap, Emoji и DialogTransmitter; устранение дублирующегося кода

Ramblball committed:
refactor, doc, style(view): форматирование всех классов представления

Ramblball committed:
feature(app): добавление настроек для работы с heroku

Ramblball committed:
refactor(view):взятие токена из переменных среды в TelegramBot

Ramblball committed:
feature (database): настройка hibernate для работы с heroku

Ramblball committed:
Merge branch 'heroku' into Task3

Ramblball committed:
fix(view,database):  исправление минорных багов

Обнуление состояния пользователя при обратном переходе.
Выдача корректного сообщения об ошибке при попытке повторно подписаться на мероприятие

Ramblball committed:
refactor(app): исправление текстовых опечаток

Tempestmon committed:
refactor(app): Исправление текстовых опечаток

Ramblball committed:
docs(app): исправление опечаток в javadoc

Ramblball committed:
refactor,docs(view): замена классов команд и диалогов на enum

Tempestmon committed:
docs(app): обновление информации по деплою на Хероку

Ramblball committed:
Merge pull request #2 from Ramblball/Task3

Task3

Ramblball committed:
feature(app): добавление возможности получить случаное мероприятие

Ramblball committed:
feature(database): удаление прошедших мероприятий

Ramblball committed:
feature(app): добавление ограничения количества участников мероприятия

Ramblball committed:
feature(app): добавление возможности задать место проведения на карте

Tempestmon committed:
feature(app): добавление возможности просмотра местоположения мероприятия

Tempestmon committed:
feature(all): добавление отображения дней до начала мероприятия

Tempestmon committed:
fix(app): исправление отрицательной даты

Tempestmon committed:
fix(app): замена ссылки на карту местоположения вместо запроса

Tempestmon committed:
fix(app): исправление нерабочей ссылки

Ramblball committed:
refactor(view, controller): перенос Validate на уровень представления

Ramblball committed:
redactor(app): замена  TimerTask на ScheduledExecutorService

Tempestmon committed:
feature(app): добавление возможности поиска GIF по запросу пользователя

Ramblball committed:
refactor(app): исправление pull-request

Ramblball committed:
refactor(database): замена abstract class DAO на интерфейс DAO

Ramblball committed:
refactor(view): получение GIF по описанию мероприятияя

Ramblball committed:
chor(app): добавлние .uml диаграмы классов

Tempestmon committed:
Merge pull request #3 from Ramblball/Task4

Task#4
